packages/wdio-crossbrowsertesting-service/src/service.ts

```diff
@@ -0,0 +1,218 @@
+import got from 'got'
+import logger from '@wdio/logger'
+
+const log = logger('@wdio/crossbrowsertesting-service')
+const jobDataProperties = ['name', 'tags', 'public', 'build', 'extra']
+
+export default class CrossBrowserTestingService implements WebdriverIO.ServiceInstance {
+    private _browser!: WebdriverIO.BrowserObject | WebdriverIO.MultiRemoteBrowserObject
```

Comment:
Why do we need the `!`?
```suggestion
    private _browser: WebdriverIO.BrowserObject | WebdriverIO.MultiRemoteBrowserObject
```

Reply:
Had to add these since we are not initialising the `browser` object in the constructor. If I make it optional, then in so many places we have to handle the `possibly undefined` error

Reply:
Gotcha, let's try to minimise this. As mentioned above if we move all the assignments from `beforeSession` into the constructor we might be able to get rid of some. For `browser` it is fine to keep `!` as we know it will be provided by the system.